app/logical/sources/strategies/hentai_foundry.rb

```diff
@@ -0,0 +1,116 @@
+# Image URLs
+#
+# * https://pictures.hentai-foundry.com/a/Afrobull/795025/Afrobull-795025-kuroeda.png
+#
+# Page URLs
+#
+# * https://www.hentai-foundry.com/pictures/user/Afrobull/795025/kuroeda
+# * https://www.hentai-foundry.com/pictures/user/Afrobull/795025o
+#
+# Preview URLs
+#
+# * https://thumbs.hentai-foundry.com/thumb.php?pid=795025&size=350
+#
+# Profile URLs
+#
+# * https://www.hentai-foundry.com/user/kajinman/profile
+# * https://www.hentai-foundry.com/pictures/user/kajinman
+# * https://www.hentai-foundry.com/pictures/user/kajinman/scraps
+# * https://www.hentai-foundry.com/user/J-likes-to-draw/profile
+
+module Sources
+  module Strategies
+    class HentaiFoundry < Base
+      BASE_URL =    %r!\Ahttps?://(?:www\.)?hentai-foundry\.com!i
+      PAGE_URL =    %r!#{BASE_URL}/pictures/user/(?<artist_name>[\w-]+)/(?<illust_id>\d+)(?:/[\w-]*)?\z!i
+      PROFILE_URL = %r!#{BASE_URL}/(?:pictures/)?user/(?<artist_name>[\w-]+)(?:/[a-z]*)?\z!i
+      IMAGE_URL =   %r!\Ahttps?://pictures\.hentai-foundry\.com/\w/(?<artist_name>[\w-]+)/(?<illust_id>\d+)/[\w-]+\.\w+\z!i
+
+      def domains
+        ["hentai-foundry.com"]
+      end
+
+      def site_name
+        "Hentai Foundry"
+      end
+
+      def image_urls
+        image = page&.search("#picBox img")
+
+        return [] unless image
+
+        image.to_a.map { |img| "https:#{img["src"]}" }
+      end
+
+      def preview_urls
+        image_urls.map do |img|
+          "https://thumbs.hentai-foundry.com/thumb.php?pid=#{img[IMAGE_URL, :illust_id]}&size=250"
+        end
+      end
+
+      def page_url
+        return nil if illust_id.blank? || artist_name.blank?
+        "https://www.hentai-foundry.com/pictures/user/#{artist_name}/#{illust_id}"
+      end
+
+      def page
+        return nil if page_url.blank?
+
+        agent.get(page_url)
+      rescue Mechanize::ResponseCodeError => e
+        return nil if e.response_code.to_i == 404
+        raise
+      end
+
+      def tags
+        tags = page&.search(".boxbody [rel='tag']")&.to_a
+
+        return [] unless tags
+
+        tags.map(&:text)
```

Comment:
shouldn't it return an array of `[tag_name, tag_page]` for each tag, like the other strategies, e.g. [pixiv](https://github.com/danbooru/danbooru/blob/86e3bd8fbc6b11cee543e7cee3747b907456c9af/app/logical/sources/strategies/pixiv.rb#L222)

Reply:
Correct, my mistake.